[R] (nova) Add foreignField when defining relations

0ctanium opened the pull request:
In the LinkCollectionOptions, I added a new property named `foreignField` aside from the `field` property. 

It allows customization of the foreign field of a nova relation instead of the hardcoded `_id` foreign field

theodorDiaconu commented:
Looks merge-able just add a test and make sure to also test the query from the inverse side too.

0ctanium commented:
Yes, I was planning to do it tonight. But I keep getting the 'Cannot find name 'it'. Do you need to install type definitions for a test runner? Try `npm i --save-dev @types/jest` or `npm i --save-dev @types/mocha`' error whilst '@types/jest' is installed 🤔

So if you already had it on the repo can you tell me how you fixed it? Else I'll look into it myself

0ctanium commented:
I managed to make the tests work!
I added the possibility of using a custom env var for the test database. And still fall back with the default value.

I added 4 tests for the foreign key oneToOne & manyToMany for classic and nested foreign fields.

theodorDiaconu commented on packages/nova/DOCUMENTATION.md:
````diff
@@ -259,6 +259,22 @@ addLinks(Users, {
+
+You can decide which foreign field to use by specifying `foreignField` in the relationship configuration.
+
+```typescript
````
In this scenario we would use Comments (direct link via userId) to do this. Let's give a real world example like the stripe thingie because it helps the user formulate a mental model faster.

Also how would this work with the 'inversed side' we need to add something about that too.

0ctanium replied:
Okay, I'll use the stripe example.

And the 'inversed side' has no changes from nominal usage. So do you think it is really necessary to write "to use foreign fields from the inversed side, change nothing" ? 😅

theodorDiaconu replied:
yes, it is reasonable to have the inverse side as well.
